app/code/VConnect/OrderVolume/Model/Order/Item/OrderItemsVolumeCalculator.php

```diff
@@ -0,0 +1,95 @@
+<?php
+declare(strict_types=1);
+
+namespace VConnect\OrderVolume\Model\Order\Item;
+
+use Magento\Catalog\Model\Product;
+use Magento\Sales\Api\Data\OrderItemInterface;
+use VConnect\OrderVolume\Model\Product\Config as ProductConfig;
+
+class OrderItemsVolumeCalculator
+{
+    private array $orderItemsVolumeData = [];
+
+    public function calculate(array $orderItems): void
+    {
+        /** @var OrderItemInterface $orderItem */
+        foreach ($orderItems as $orderItem) {
+            $this->setCalculationResult((int)$orderItem->getItemId(), $this->getOrderItemVolume($orderItem));
+        }
+    }
+
+    /**
+     * @param OrderItemInterface $orderItem
+     * @return float|int
+     * @throws \Exception
+     */
+    private function getOrderItemVolume(OrderItemInterface $orderItem): float|int
+    {
+        $orderItemQuantity = $orderItem->getQtyOrdered();
+        $productVolume = $this->getProductVolume($orderItem);
+        if ($this->isFloat($productVolume)) {
+            $productVolume = (float)$productVolume;
+        } else {
+            $productVolume = (int)$productVolume;
+        }
+
+        return $this->calculateOrderItemVolume($orderItemQuantity, $productVolume);
+    }
+
+    /**
+     * @param OrderItemInterface $orderItem
+     * @return mixed
+     * @throws \Exception
+     */
+    private function getProductVolume(OrderItemInterface $orderItem)
+    {
+        /** @var Product $product */
+        $product = $orderItem->getProduct();
+
+        /** @var \Magento\Framework\Api\AttributeInterface $customAttributeInterface */
+        $customAttributeInterface = $product->getCustomAttribute(ProductConfig::PRODUCT_VOLUME);
+        if ($customAttributeInterface === null) {
+            throw new \Exception(
```

Comment:
Are you sure all products must have volume assigned? what if on your site will be a product like a Virtual gift card? It means customer can't place order with such a product?

First option: just return nothing if volume doesn't exist. 
Second one: catch expections in your methods and log information to debug.log like 'Product SKU: $sku doesn't have volume assigned'